integrations/tests/test_maltiverse.py

```diff
@@ -140,6 +140,21 @@ def test_get_by_sha1():
         mock_get.assert_called_once_with(f'https://api.maltiverse.com/sample/sha1/{example_sample}')
 
 
+@pytest.mark.parametrize('url,expected', [
+    ('http://example.com', True),
+    ('https://example.com:8080', True),
+    ('https://api.maltiverse.com', True),
+    ('http://example.com/<script>', False),
+    ('https://192.168.0.1:invalid', False),
+    ('malformed://example.com', False),
+    ('http://', False)
```

Comment:
Let's add some URLs from the [issue](https://github.com/python/cpython/issues/103848) that reported the CVE to be sure we reject them if there are changed in the future:

```
https://user:some]password[@host.com
https://user:some%5Dpassword%5B@host.com
```

Reply:
Added in 76c13c142be9b4c40c3f97316fef6871ccc554cc.

```console
(env) federamos@pop-os:~/Documents/Wazuh/Repositories/wazuh$ PYTHONPATH=/home/federamos/Documents/Wazuh/Repositories/wazuh/framework/:/home/federamos/Documents/Wazuh/Repositories/wazuh/api python3 -m pytest integrations/
========================================================================================================== test session starts ==========================================================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/federamos/Documents/Wazuh/Repositories/wazuh/integrations
configfile: pytest.ini
plugins: asyncio-0.18.1, html-2.1.1, metadata-3.1.1, cov-4.1.0, tavern-1.23.5, anyio-4.1.0, trio-0.8.0
asyncio: mode=auto
collected 131 items                                                                                                                                                                                                                     

integrations/tests/test_maltiverse.py .......................................................................                                                                                                                     [ 54%]
integrations/tests/test_pagerduty.py ..........                                                                                                                                                                                   [ 61%]
integrations/tests/test_shuffle.py ..................                                                                                                                                                                             [ 75%]
integrations/tests/test_slack.py ..........                                                                                                                                                                                       [ 83%]
integrations/tests/test_virustotal.py ......................                                                                                                                                                                      [100%]

========================================================================================================== 131 passed in 2.92s ==========================================================================================================
```